### ISSUE TITLE
"Tested by data point" in CodeLens is failing because RemoteCodeLensReferencesService is returning a collection containing nulls

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#20131]
HIt this while debugging VS.

No idea who owns the RemoveCodeLensReferencesSerivice, but it's FindReferencesMethodAsync is returning an enumerable with null values:

![image](https://user-images.githubusercontent.com/1103906/26962787-009d28d2-4d2c-11e7-957d-d75fea740d07.png)

This is causing this method in TestedByDataPoint.FindMethodReferences to fail:

``` C#
        private async Task<IList<MethodDescriptor>> FindMethodReferences(CancellationToken token)
        {
            {
                token.ThrowIfCancellationRequested();

                DocumentId documentId = await this.TryGetDocumentId(token).ConfigureAwait(false);

                var document = this.integrationService.GetSolution().GetDocument(documentId);

                var referenceMethods = await this.referencesService.FindReferenceMethodsAsync(
                    this.integrationService.GetSolution(),
                    documentId, 
                    await this.Descriptor.SyntaxNode.GetCurrentSyntaxNodeAsync(document).ConfigureAwait(false),
                    token).ConfigureAwait(false);

                return referenceMethods.Select(rm => new MethodDescriptor(rm.FullName, rm.FilePath)).ToList(); // NullReferenceException here due to "rm" being null
            }
        }
```

**Comment by davkean** [comment]
This occurs happening opening this file: https://github.com/dotnet/project-system/blob/master/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/CrossTargetSubscriptionHostBase.cs.

**Comment by davkean** [comment]
Basically this means that you get no _tested by_ data points at all in the ProjectSystem.sln solution which is probably related.

**Comment by davkean** [comment]
I cannot find code for RemoveCodeLensReferencesService _anywhere_.

**Comment by CyrusNajmabadi** [comment]
Is it not this code: https://github.com/dotnet/roslyn/blob/master/src/VisualStudio/Core/Next/CodeLens/RemoteCodeLensReferencesService.cs

**Comment by davkean** [comment]
Oh weird, could not find that.

**Comment by Pilchie** [comment]
@panopticoncentral any ideas off the top of your head?

**Comment by panopticoncentral** [comment]
If CodeLensReferencesService.GetMethodDescriptorAsync fails for any reason, it returns null. So there's probably something about that file that's causing that method to fail.

So, really, two issues: a) Why is the method failing?, and b) FindReferenceMethodsAsync should filter out null after the Select so that it doesn't take down all test finding.

**Comment by panopticoncentral** [comment]
I took a little bit of a look at it. It looks like we've found a reference that's an override, so the reference is the name of the overriding method. We call SemanticModel.GetEnclosingSymbol looking for the enclosing method but in this case we don't get one, and GetMethodDecscriptoAsync just returns null.

Probably we should be handling this correctly, as it means we're likely not showing references in CodeLens that are overrides (or maybe we didn't show them in the past?).

In any case, we should definitely protect against the nulls escaping.
